h2o-core/src/main/java/hex/Model.java

```diff
@@ -1535,7 +1535,7 @@ else if (isWeights && computeMetrics) {
             if( isResponse && vec.domain() != null && ds.length == domains[i].length+vec.domain().length )
               throw new IllegalArgumentException("Test/Validation dataset has a categorical response column '"+names[i]+"' with no levels in common with the model");
             if (ds.length > domains[i].length)
-              msgs.add("Test/Validation dataset column '" + names[i] + "' has levels not trained on: " + ArrayUtils.toStringQuotedElements(Arrays.copyOfRange(ds, domains[i].length, ds.length)));
+              msgs.add("Test/Validation dataset column '" + names[i] + "' has levels not trained on: " + ArrayUtils.toStringQuotedElements(Arrays.copyOfRange(ds, domains[i].length, ds.length), 20));
```

Comment:
I suggest also printing how many unseen levels are there in total.

Reply:
good idea, maybe I can put this in the ellipsis: 
```
["a", "b", ...8 more..., "k", "l"]
```